addons/pos_online_payment_self_order/controllers/payment_portal.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+
+from odoo.http import request
+from odoo.addons.pos_online_payment.controllers.payment_portal import PaymentPortal
+
+class PosSelfOrderPaymentPortal(PaymentPortal):
+
+    def _render_pay_confirmation(self, rendering_context):
+        if rendering_context.get('state') == 'success':
+            request.env['pos_preparation_display.order'].sudo().process_order(rendering_context.get('pos_order_id'))
```

Comment:
I don't think is the right place to call `pos_preparation_display` because `pos_online_payment_self_order` do not depend of it